CharlieBackend.AdminPanel/Services/CourseService.cs

```diff
@@ -19,16 +21,22 @@ public class CourseService: ICourseService
 
         private readonly IMapper _mapper;
 
-        public CourseService(IApiUtil apiUtil, IOptions<ApplicationSettings> config, IMapper mapper)
+        private readonly string _accessToken;
+
+        public CourseService(IApiUtil apiUtil, IOptions<ApplicationSettings> config, IMapper mapper, 
+            IHttpContextAccessor httpContextAccessor, IDataProtectionProvider provider)
         {
             _apiUtil = apiUtil;
             _config = config;
             _mapper = mapper;
+
+            IDataProtector protector = provider.CreateProtector(_config.Value.Cookies.SecureKey);
+            _accessToken = protector.Unprotect(httpContextAccessor.HttpContext.Request.Cookies["accessToken"]);
```

Comment:
What if it fails to decypher access token obtained from inside the cookie? (And why is it encrypted in the first place? What is the point of encrypting JWT?)

If `Unprotect()` method throws an [exception](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.dataprotection.idataprotector.unprotect?view=aspnetcore-5.0#exceptions), then exception handler here
https://github.com/ita-social-projects/WhatBackend/blob/0f78082fd6c8f03ede68ab971f5a5d0371028fc6/CharlieBackend.AdminPanel/Middlewares/ExceptionHandleMiddleware.cs#L33
will end up writing *HTTP 500: Internal Server Error* response to a client.

Reply:
Yes, an undescribed error would occure. But it is a common way to obtain tokey through all the Services in AdminPanel.
I've suggested this changes to continue work with Views. Which needs a solid set of Service interfaces.
I think, the inner realisation of All Services should be enhaced in futher development at one pull request.